tests/tape/test_tape.py

```diff
@@ -804,11 +804,11 @@ def test_array_parameter(self):
             qml.Rot(params[1], params[2], params[3], wires=0)
 
         assert tape.num_params == len(params)
-        assert tape.get_parameters() == params
 
-        b = np.array([0, 1, 0, 0])
+        b = np.array([0.0, 1.0, 0.0, 0.0])
```

Comment:
Why this is changed?

Reply:
The problem is that now I preprocess in StatePrep and it is transforming it by default into a float array
If I leave it as integers, it gives me problems in differentiation 🤔